src/app/core/models/user.model.ts

```diff
@@ -0,0 +1,8 @@
+export interface IUser {
+    userId?: string;
+    email?: string;
+    password?: string;
+    name?: string;
+    surname?: string;
+    role?: string;
```

Comment:
What it the reason to make all fields optional?

Reply:
Probably, I made a mistake :)